carapace-server/src/main/java/org/carapaceproxy/core/HttpProxyServer.java

```diff
@@ -717,6 +723,10 @@ private void applyDynamicConfiguration(ConfigurationStore newConfigurationStore,
                 proxyRequestsManager.reloadConfiguration(newConfiguration, newBackends.values());
             }
 
+            if(currentConfiguration.isMaintenanceModeEnabled() != newConfiguration.isMaintenanceModeEnabled()) {
```

Comment:
I think this is useless, because this code is run only when a certificare is uploaded from api and in thad case the mantainance mode shouldn't change